test_conformance/half/Test_vStoreHalf.cpp

```diff
@@ -562,151 +632,230 @@ int Test_vStoreHalf_private( cl_device_id device, f2h referenceFunc, d2h doubleR
             }
 
-            doubleKernels[ vectorSize ][1] = clCreateKernel( doublePrograms[ vectorSize ][1], "test", &error );
-            if( NULL == kernels[vectorSize][1] )
+            doubleKernels[vectorSize][1] =
+                clCreateKernel(doublePrograms[vectorSize][1], "test", &error);
+            if (NULL == kernels[vectorSize][1])
             {
                 gFailCount++;
-                vlog_error( "\t\tFAILED -- Failed to create double private kernel. (%d)\n", error );
+                vlog_error("\t\tFAILED -- Failed to create double private "
+                           "kernel. (%d)\n",
+                           error);
                 return error;
             }
 
-            if(g_arrVecSizes[vectorSize] == 3) {
-                doublePrograms[vectorSize][2] = MakeProgram( device, double_source_local_store_v3, sizeof(double_source_local_store_v3) / sizeof( double_source_local_store_v3[0]) );
-            } else {
-                doublePrograms[vectorSize][2] = MakeProgram( device, double_source_local_store, sizeof(double_source_local_store) / sizeof( double_source_local_store[0]) );
+            if (g_arrVecSizes[vectorSize] == 3)
+            {
+                doublePrograms[vectorSize][2] =
+                    MakeProgram(device, double_source_local_store_v3,
+                                sizeof(double_source_local_store_v3)
+                                    / sizeof(double_source_local_store_v3[0]));
+            }
+            else
+            {
+                doublePrograms[vectorSize][2] =
+                    MakeProgram(device, double_source_local_store,
+                                sizeof(double_source_local_store)
+                                    / sizeof(double_source_local_store[0]));
             }
-            if( NULL == doublePrograms[ vectorSize ][2] )
+            if (NULL == doublePrograms[vectorSize][2])
             {
                 gFailCount++;
                 return -1;
             }
 
-            doubleKernels[ vectorSize ][2] = clCreateKernel( doublePrograms[ vectorSize ][2], "test", &error );
-            if( NULL == kernels[vectorSize][2] )
+            doubleKernels[vectorSize][2] =
+                clCreateKernel(doublePrograms[vectorSize][2], "test", &error);
+            if (NULL == kernels[vectorSize][2])
             {
                 gFailCount++;
-                vlog_error( "\t\tFAILED -- Failed to create double local kernel. (%d)\n", error );
+                vlog_error("\t\tFAILED -- Failed to create double local "
+                           "kernel. (%d)\n",
+                           error);
                 return error;
             }
         }
     } // end for vector size
 
+    const char *reset[] = {
+        "__kernel void reset( __global float *p, __global ushort *f,\n"
+        "                   uint extra_last_thread)\n"
+        "{\n"
+        "   size_t i = get_global_id(0);\n"
+        "   *(f + i) = 0xdeaddead;"
```

Comment:
I'm seeing the following warning with on this line of kernel code
```
kernel.opencl:5:15: warning: implicit conversion from 'unsigned int' to 'ushort' (aka 'unsigned short') changes value from 3735936685 to 57005
````

Reply:
Fixed.

I wonder if this could be sped up even more by storing 32-bit ints instead of 16-bit ints...